features/lorawan/LoRaWANStack.h

```diff
@@ -459,21 +450,44 @@ class LoRaWANStack: private mbed::NonCopyable<LoRaWANStack> {
      *
      * @param  port            The event to be sent.
      */
-    void send_automatic_uplink_message(const uint8_t port);
+    void send_automatic_uplink_message(uint8_t port);
+
+    /**
+     * TX interrupt handlers and corresponding processors
+     */
+    void tx_interrupt_handler(void);
+    void tx_timeout_interrupt_handler(void);
+    void process_transmission(void);
+    void process_transmission_timeout(void);
+
+    /**
+     * RX interrupt handlers and corresponding processors
+     */
+    void rx_interrupt_handler(const uint8_t *payload, uint16_t size, int16_t rssi,
+                              int8_t snr);
+    void rx_timeout_interrupt_handler(void);
+    void rx_error_interrupt_handler(void);
+    void process_reception(const uint8_t *payload, uint16_t size, int16_t rssi,
+                           int8_t snr);
+    void process_reception_timeout(bool is_timeout);
+
+    int convert_to_msg_flag(const mcps_type_t type);
 
 private:
     LoRaMac _loramac;
-    loramac_primitives_t LoRaMacPrimitives;
-
+    radio_events_t radio_events;
     device_states_t _device_current_state;
     lorawan_app_callbacks_t _callbacks;
     lorawan_session_t _lw_session;
     loramac_tx_message_t _tx_msg;
     loramac_rx_message_t _rx_msg;
     uint8_t _num_retry;
+    uint32_t _ctrl_flags;
```

Comment:
Okay, so this isn't the enum? Because the compiler's correctly stopping you doing the bit manipulations, I guess?

Reply:
Flags are now made Macros for clarity.